src/Simplic.Flow.Node/ActionNode/Base/SqlNode.cs

```diff
@@ -0,0 +1,227 @@
+﻿using CommonServiceLocator;
+using Simplic.Flow.Event;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Simplic.Sql;
+using Dapper;
+using Microsoft.SqlServer.Server;
+
+namespace Simplic.Flow.Node
+{
+    /// <summary>
+    /// Clear pin node
+    /// </summary>
+    [ActionNodeDefinition(DisplayName = "Sql Select Statement", Name = nameof(SqlNode), Category = "Common")]
+    public class SqlNode : ActionNode
+    {
+        private readonly ISqlService sqlService = CommonServiceLocator.ServiceLocator.Current.GetInstance<ISqlService>();
+
+        /// <summary>
+        /// Gets all object from the sql statement.
+        /// </summary>
+        /// <param name="sql">The string with the select statement.</param>
+        /// <returns>Returns a enumerable of of all objects of the table.</returns>
+        private IEnumerable<object> GetAllTableData(string sql)
+        {
+            return sqlService.OpenConnection((connection) =>
+            {
+                return connection.Query<object>(sql);
+            });
+        }
+
+        /// <summary>
+        /// Execute.
+        /// </summary>
+        /// <param name="runtime">Runtime instance.</param>
+        /// <param name="scope">Scope instance.</param>
+        /// <returns>True or false, if the table name is not specified.</returns>
+        public override bool Execute(IFlowRuntimeService runtime, DataPinScope scope)
+        {
+            try
+            {
+                var eventService = ServiceLocator.Current.GetInstance<IFlowEventService>();
+
+                var tableName = scope.GetValue<string>(InPinTableName);
+
+                if (string.IsNullOrWhiteSpace(tableName))
+                {
+                    Console.WriteLine($"No table name is set in {nameof(SqlNode)}");
+                    return false;
+                }
+
+                var column1 = scope.GetValue<string>(InPinColumn01);
+                var column2 = scope.GetValue<string>(InPinColumn02);
+                var column3 = scope.GetValue<string>(InPinColumn03);
+
+                var parameter1 = scope.GetValue<string>(InPinParameter01);
+                var parameter2 = scope.GetValue<string>(InPinParameter02);
+                var parameter3 = scope.GetValue<string>(InPinParameter03);
+
+                // Start with the basic SELECT statement.
+                string sql = $"SELECT * FROM {tableName}";
+
+                // Create the WHERE condition depending on the existing parameters.
+                List<string> conditions = new List<string>();
+
+                if (!string.IsNullOrEmpty(column1) && !string.IsNullOrEmpty(parameter1))
```

Comment:
IsNullOrWhitespace for the column might be better.
maybe the parameter can be allowed to be null when the user wants to filter for null values.
Also what happens when the user wants to filter for bools or numbers, that is currently not possible.
And there is a possibillity for sql injections be done.